sdk/core/Azure.Core/src/ResourceIdentifier.cs

```diff
@@ -470,6 +469,45 @@ public override int GetHashCode()
             return left is null ? right is null : left.CompareTo(right) >= 0;
         }
 
+        /// <summary>
+        /// Converts the string representation of a ResourceIdentifier to the equivalent <see cref="ResourceIdentifier"/> structure.
```

Comment:
```suggestion
        /// Parses the string representation of an Azure resource identifier to the equivalent <see cref="ResourceIdentifier"/> structure.
```

Reply:
@christothes for all of these comments I simply copied Guid.Parse, let me know if copying that is the right way to go or make the changes you suggest here.

Reply:
ok, I will leave now for consistency we can change later if we feel its necessary